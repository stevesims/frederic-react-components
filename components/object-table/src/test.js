import React from 'react';
import { mount, shallow } from 'enzyme';

import ObjectTable from '.';

const fields = [
  { key: 'one', heading: 'One' },
  { key: 'two', heading: 'Two', transform: () => 'two' },
];

const object = { one: 'test', two: 'test' };

describe('ObjectTable', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ObjectTable />);
    expect(wrapper.exists()).toBe(true);
  });
  
  describe('renders table data', () => {
    it('renders expected table data cells', () => {
      const wrapper = mount(<ObjectTable fields={fields} object={object} />); 
      expect(wrapper.find('Table').prop('rows')).toEqual([['One', 'test'], ['Two', 'two']]);
    });

    it('omits rows in table for null values', () => {
      const fieldsWithTransform = [
        { key: 'one', heading: 'one' },
        { key: 'two', heading: 'two', transform: () => null },
        { key: 'three', heading: 'three'},
      ];
      const objectWithNull = { one: 'test', two: 'test', three: null };
      const wrapper = mount(<ObjectTable fields={fieldsWithTransform} object={objectWithNull} skipEmptyValues />);
      expect(wrapper.find('tr').length).toBe(1);
    });
  
    it('renders nothing when rows have no values and hideWithNoValues is true', () => {
      const objectWithNoValues = {};
      const wrapper = mount(<ObjectTable fields={fields} object={objectWithNoValues} hideWithNoValues />);   
      expect(wrapper.html()).toBe(null);
    });
  });

  describe('responds as expected to additional props', () => {
    it('optionally renders a heading for a table', () => {
      const wrapper = shallow(<ObjectTable fields={fields} object={object} title="Heading" />);
      expect(wrapper.contains('Heading')).toBe(true);
    });
  });

  it('matches snapshot', () => {
    const wrapper = mount(<ObjectTable fields={fields} object={object} />);
    expect(wrapper).toMatchSnapshot();
  });
});

