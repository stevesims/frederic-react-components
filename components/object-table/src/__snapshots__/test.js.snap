// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectTable matches snapshot 1`] = `
.emotion-10 {
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-10 {
    margin-bottom: 15px;
  }
}

.emotion-8 {
  display: table;
  font-family: "nta",Arial,sans-serif;
  width: 100%;
  table-layout: fixed;
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-8 {
    margin-bottom: 15px;
  }
}

.emotion-0 {
  border-bottom: 1px solid #bfc1c3;
  display: table-cell;
  font-size: 14px;
  padding: 15px 4px;
  vertical-align: top;
  font-weight: bold;
  text-align: left;
  width: 25%;
}

.emotion-0:first-child {
  padding: 15px 4px 15px 0;
}

.emotion-0:last-child {
  padding: 15px 0 15px 4px;
}

.emotion-2 {
  border-bottom: 1px solid #bfc1c3;
  display: table-cell;
  font-size: 14px;
  padding: 15px 4px;
  vertical-align: top;
}

.emotion-2:first-child {
  padding: 15px 4px 15px 0;
}

.emotion-2:last-child {
  padding: 15px 0 15px 4px;
}

<ObjectTable
  fields={
    Array [
      Object {
        "heading": "One",
        "key": "one",
      },
      Object {
        "heading": "Two",
        "key": "two",
        "transform": [Function],
      },
    ]
  }
  hideWithNoValues={false}
  object={
    Object {
      "one": "test",
      "two": "test",
    }
  }
  skipEmptyValues={false}
  title={null}
>
  <Styled(Table)
    flexibleColumns={false}
    nameByRow={true}
    names={
      Array [
        "one",
        "two",
      ]
    }
    rowIncludesHeading={true}
    rows={
      Array [
        Array [
          "One",
          "test",
        ],
        Array [
          "Two",
          "two",
        ],
      ]
    }
    titles={null}
  >
    <Table
      className="emotion-10"
      flexibleColumns={false}
      nameByRow={true}
      names={
        Array [
          "one",
          "two",
        ]
      }
      rowIncludesHeading={true}
      rows={
        Array [
          Array [
            "One",
            "test",
          ],
          Array [
            "Two",
            "two",
          ],
        ]
      }
      titles={null}
    >
      <TableContainer
        className="emotion-10"
        flexibleColumns={false}
      >
        <table
          className="emotion-8 emotion-9"
        >
          <tbody>
            <tr
              key="0"
            >
              <TableHeading
                columnCount={2}
                key="0"
                name="one"
                rowHeading={true}
              >
                <th
                  className="emotion-0 emotion-1"
                  name="one"
                >
                  One
                </th>
              </TableHeading>
              <TableData
                key="1"
                name="one"
              >
                <td
                  className="emotion-2 emotion-3"
                  name="one"
                >
                  test
                </td>
              </TableData>
            </tr>
            <tr
              key="1"
            >
              <TableHeading
                columnCount={2}
                key="0"
                name="two"
                rowHeading={true}
              >
                <th
                  className="emotion-0 emotion-1"
                  name="two"
                >
                  Two
                </th>
              </TableHeading>
              <TableData
                key="1"
                name="two"
              >
                <td
                  className="emotion-2 emotion-3"
                  name="two"
                >
                  two
                </td>
              </TableData>
            </tr>
          </tbody>
        </table>
      </TableContainer>
    </Table>
  </Styled(Table)>
</ObjectTable>
`;
